src/xdp-app-info-registry.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright © 2025 Red Hat, Inc
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+
+#include "xdp-app-info.h"
+
+#include "xdp-app-info-registry.h"
+
+struct _XdpAppInfoRegistry
+{
+  GObject parent_instance;
+
+  GHashTable *app_infos; /* unique dbus name -> app info */
+  GMutex app_infos_mutex;
+};
+
+G_DEFINE_FINAL_TYPE (XdpAppInfoRegistry,
+                     xdp_app_info_registry,
+                     G_TYPE_OBJECT)
+
+static void
+xdp_app_info_registry_dispose (GObject *object)
+{
+  XdpAppInfoRegistry *registry = XDP_APP_INFO_REGISTRY (object);
+
+  if (registry->app_infos)
```

Comment:
This condition is redundant.

Reply:
Yeah almost left this comment myself... Certainly a gotcha